### DISCUSSION TITLE
[cairomm][pangomm]The include pathes are wrong

### DISCUSSION THREAD

**Discussion by yonaka** [microsoft/vcpkg#19328] [Q&A]
**Describe the bug**

Includes below do not work.
```
#include <cairomm/cairomm.h>
#include <pangomm.h>
```

Ones below, which are unusual, do work.
```
#include <cairomm-1.16/cairomm/cairomm.h>
#include <pangomm-2.48/pangomm.h>
```

**Environment**
- OS: Windows 10 20H2
- Compiler: Visual Studio 2019 16.10.3

**To Reproduce**
See the description above.

**Expected behavior**
The code below should not emit any error.
```
#include <cairomm/cairomm.h>
#include <pangomm.h>
```

**Additional context**
A page below reads the include should be `#include <pangomm.h>`
https://developer.gnome.org/pangomm/stable/

I suppose some github searches indicate the include of cairomm.h should be `#include <cairomm/cairomm.h>`.

In addition, please check pango, glibmm and geomm packages.

**Comment by yonaka** [comment]
Possibly relating to: #18760
At least, the minor version numbers should not be included in the path.

**Comment by NancyLi1013** [comment]
Hi @yonaka 

Thanks for posting this issue. 
 For the include path, please see the comments here https://github.com/microsoft/vcpkg/issues/18760#issuecomment-872812050.

I think this is the same problem with which in #18760. You should add the path to your project manually. For example, 


![image](https://user-images.githubusercontent.com/46708020/124534632-3e0cc280-de47-11eb-8bd7-835f40b809ea.png)

Then you can include the headers without the directory that has the version, such as `cairomm-1.16` and `pangomm-2.48`.

**Comment by yonaka** [comment]
@NancyLi1013 
Thank you for the workaround.

It means each time the library is updated we will have to change the include directory project setting, e.g. `cairomm-1.62` to `cairomm-1.63`. This is unreasonable.
These directory names should be `cairomm` or `cairomm-1` rather than `cairomm-1.62` and `pangomm` or `pangomm-2` rather than `pangomm-2.48`, I suppose.
If we were to need to be able to use cairomm version 1.x and 2.x simultaneously (in future), it should be `cairomm-1` rather than `cairomm`.

**Comment by Neumann-A** [comment]
@yonaka If you would use CMake or Meson you wouldn't have those problems since the pc file contains the info for the include folders. The MSBuild integration is not clever enough to setup "usage" include folders. The folder structure itself is given by upstream.

**Comment by yonaka** [comment]
I asume this is a correct ExternalIncludePath setting to compile `#include <pangomm.h>` successfully with MSBuild (Visual Studio):

`$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\cairo\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\harfbuzz\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\pango-1.0\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\sigc++-3.0\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\giomm-2.68\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\glibmm-2.68\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\cairomm-1.16\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\pangomm-2.48\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\include\glib-2.0\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\lib\pangomm-2.48\include\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\lib\giomm-2.68\include\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\lib\sigc++-3.0\include\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\lib\cairomm-1.16\include\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\lib\glib-2.0\include\;$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows\lib\glibmm-2.68\include\;$(ExternalIncludePath)`

This is annoying.

**Comment by nakag2020** [comment]
Also possibly relating to: #17702 #17537

**Comment by Neumann-A** [comment]
@yonaka You probably can shorten it by using `$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)` instead of `$(VCPkgRoot)\installed\$(LibrariesArchitecture)-windows`. The former also works with custom triplets.

**Comment by yonaka** [comment]
@Neumann-A Thank you for the info.

`$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\cairo\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\harfbuzz\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\pango-1.0\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\sigc++-3.0\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\giomm-2.68\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\glibmm-2.68\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\cairomm-1.16\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\pangomm-2.48\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\include\glib-2.0\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\lib\pangomm-2.48\include\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\lib\giomm-2.68\include\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\lib\sigc++-3.0\include\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\lib\cairomm-1.16\include\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\lib\glib-2.0\include\;$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)\lib\glibmm-2.68\include\;$(ExternalIncludePath)`

Aside from the length of the settings, it is tedious to have to figure out by trial and error what paths are needed to make just a `#include <pangomm.h>` compiled successfully.

**Comment by yonaka** [comment]
Correction: `$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)..` does not work somehow. This seems OK on the property window but does not compile.

**Comment by Neumann-A** [comment]
use only `$(_ZVcpkgCurrentInstalledDir)/include` for includes. `$(_ZVcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)/include` does not exists in for `debug` configurations.  Basically only use `$(VcpkgConfigSubdir)` for the lib subfolder

**Comment by yonaka** [comment]
Could you show me the example settings?
